### PR TITLE
Refactor pycode

### PR DESCRIPTION
**Pull request by tk0miya**
To support new syntax of python3 (ex, async, type annotations and so on), I try to replace pycode using the resources of python itself.
For example, this uses `tokenize` and `ast` module as parser components.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -19,13 +19,11 @@ include sphinx/locale/.tx/config
 recursive-include sphinx/templates *
 recursive-include sphinx/texinputs *
 recursive-include sphinx/themes *
-recursive-include sphinx/pycode/pgen2 *.c *.pyx
```
Comment (shimizukawa) [inline MANIFEST.in:22]: Awesome! 👍

**Comment by shimizukawa** [comment]
Great Job!!

However I think reviewing is not realistic.
I guess there is no other way than to verify with the actual documents which are using autodoc.

**Comment by tk0miya** [comment]
I posted "Request for testing" to sphinx-users list:
https://groups.google.com/forum/#!topic/sphinx-users/ABMyJGbOtHQ

I wait for reports for a while, and will merge this after if all problem are resolved.

**Comment by c-bata** [comment]
🎉 🎉 🎉 
The documentation of [my kobin project(which uses type annotation)](https://github.com/kobinpy/kobin) is properly built.
It is temporary available from following urls (I will remove it next month).

* top page: https://s3-ap-northeast-1.amazonaws.com/kobin-docs/html/index.html
* page built by autodoc: https://s3-ap-northeast-1.amazonaws.com/kobin-docs/html/api.html

I hope this can help you :)

**Comment by shirou** [comment]
I have checked those projects which use autodoc extention.

- [pygments](http://pygments.org/)
- [requests](http://docs.python-requests.org/en/master/)
- [supervisord](https://github.com/Supervisor/supervisor/)

I checked by `diff -r` with sphinx 1.6.3 output, it looks like there are no differences!  Awesome!

**Comment by tk0miya** [comment]
@c-bata @shirou  Thank you for reporting!

**Comment by lelit** [comment]
I tried your branch over a few modules that triggered #3611, and it worked great: it extracted documentation about all module level symbols, whereas 1.6.2 could not.
Thank you!

**Comment by tk0miya** [comment]
Thank you for confirmation! Merged!!!

**Pull request merged**